Chore/automated changelog

TylorS opened the pull request:
Add ghooks and validate-commit-msg to enforce good commit messages
Add commitizen and cz-customizable to help with creating good commit messages
Add conventional-changelog to generate changelogs from commits

TylorS committed:
chore(META): implement automatic changelog generation

Add ghooks and validate-commit-msg to enforce good commit message
Add commitizen and cz-customizable to help with creating good commit messages
Add conventional-changelog to generate changelogs from commits

TylorS committed:
chore(META): build changelog

Build initial changelog from what good commits we already have

jemmyw commented:
OK looks good. Could I get you to add something to the readme about how to set it up and run it from a development flow perspective?

How will the changelog get generated? Is that something the CI can do and commit automatically when changes are merged to release?

TylorS commented:
Sure, I can do that. We can most likely get this to build automatically on CI

TylorS committed:
docs(README): add documentation on using npm run commit

Add basic information to the README that explains how to properly make use of new tooling to provide

changelog generation.
